specification/security/_types/Privileges.ts

```diff
@@ -92,14 +93,36 @@ export class IndicesPrivileges {
   /**
    * A search query that defines the documents the owners of the role have read access to. A document within the specified indices must match this query for it to be accessible by the owners of the role.
    */
-  query?: string[] | QueryContainer
+  query?: string[] | QueryContainer | RoleTemplateQueryContainer
   /**
    * Set to `true` if using wildcard or regular expressions for patterns that cover restricted indices. Implicitly, restricted indices have limited privileges that can cause pattern tests to fail. If restricted indices are explicitly included in the `names` list, Elasticsearch checks privileges against these indices regardless of the value set for `allow_restricted_indices`.
    * @server_default false
    */
   allow_restricted_indices?: boolean
 }
 
+export class RoleTemplateQueryContainer {
```

Comment:
If this is a container class, you should mark it as [container](https://github.com/elastic/elasticsearch-specification/blob/main/docs/modeling-guide.md#container).
```suggestion
/** @variants container */
export class RoleTemplateQueryContainer {
```